compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEAttrs.td

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_AMDAIE_DIALECT_IREEAMDAIEATTRS
+#define IREE_COMPILER_AMDAIE_DIALECT_IREEAMDAIEATTRS
+
+include "iree-amd-aie/IR/AMDAIEDialect.td"
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/EnumAttr.td"
+
+def AMDAIE_MemSpace_Global : I32EnumAttrCase<"Global", 0>;
+def AMDAIE_MemSpace_Shared : I32EnumAttrCase<"Shared", 1>;
+def AMDAIE_MemSpace_Local : I32EnumAttrCase<"Local", 2>;
+def AMDAIE_MemSpaceAttr: I32EnumAttr<"AMDAIEMemSpace", "AIE Memory Space",
+  [
+    AMDAIE_MemSpace_Global,
+    AMDAIE_MemSpace_Shared,
+    AMDAIE_MemSpace_Local,
+  ]> {
+
+  let cppNamespace = "mlir::iree_compiler::AMDAIE";
+}
+
+def AMDAIE_PermLevelAttr :
+  AttrDef<AMDAIE_Dialect, "PermLevel", []>
+{
+  let mnemonic = "perm_level";
+  let parameters = (ins
+    ArrayRefParameter<"int64_t","">:$perm
+  );
+  
+  let assemblyFormat = [{
+    `[` $perm `]`
+  }];
+}
+
+def AMDAIE_PermLevelsAttr :
+  ArrayOfAttr<AMDAIE_Dialect, "PermLevels",
+    "perm_levels", "PermLevelAttr", []>
+{
+}
+
+def AMDAIE_PackingConfigPackingLevelAttr :
+  AttrDef<AMDAIE_Dialect, "PackingConfigPackingLevel", []>
+{
+  let mnemonic = "packing_config_level";
+  let parameters = (ins
+    ArrayRefParameter<"int64_t",
+        "Expected packed sizes for specified iterator dimensions">:$packedSizes,
+    ArrayRefParameter<"int64_t",
+        "Indices of pack operations need to be transposed">:$transposePackIndices,
+    ArrayRefParameter<"int64_t",
```

Comment:
I am not entirely clear why there is an indicator. Pack and unpack should always go together?

Maybe @yzhang93

Reply:
This is to indicate if the transpose target is a unpack op or a pack op. If the target is a pack op, then we need to set the upackOp as null to pass to this function https://github.com/llvm/llvm-project/blob/411554a3535e55a1436ccda80064d7a91814dc27/mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h#L1139

Reply:
Shouldnt this just be a boolean then?